chapter01/1.3 - URLify/urlify.js

```diff
@@ -0,0 +1,24 @@
+var urlify = function(str, length) {
+  // have a pointer to check from start to end
+  var strArr = str.split('');
+  var pointer = 0;
+  while (pointer < str.length) {
+    if (strArr[pointer] === ' ') {
+      // *** needs more work here, a little wierd
+      // not handling trailing spaces properly
+      for (var i = str.length - 1; i > pointer + 3; i--) {
+        strArr[i] = str[i - 2];
+      }
+      strArr[pointer] = '%';
+      strArr[pointer+1] = '2';
```

Comment:
This brings up a greater debate as to whether we should use the power of JavaScript to simplify questions like this or to asnwer them in full with the data structures used in the book. I see no need to use an array here since we could work with the power of JavaScript strings to `.replace()` and `.strip()`. Thoughts?